clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -526,10 +531,12 @@ static KafkaAdminClient createInternal(AdminClientConfig config, TimeoutProcesso
                 true,
                 apiVersions,
                 null,
+                tmi,
+                tmi != null ? new ClientSensorRegistry(tmi.metrics()) : null,
```

Comment:
I agree about `null`.

The stylistic challenge I'm running into is the creation of classes that introducing interfaces would create.

Right now there are five subclasses of `AbstractSensorRegistry`. We would first need to abstract out five interfaces from those subclasses. Then we'd need to refactor the current subclasses to provide the concrete implementation. Then we'd need to add five more dummy classes that provide the no-op implementation. So that's 10 more source files.

It just seems like a _lot_ of classes to prevent these `NullPointerException`s from happening. But maybe it's worth it 🤷‍♂️ 

I guess we could do something with reflection (`InvocationHandler`) or code generation or something to get the no-op classes down to a single source file.

Reply:
Or we could perform a more extensive refactoring and rethink the concept of the `SensorRegistry` anyway so that there aren't so many classes to begin with.

Reply:
Suggestions welcome 😉

Reply:
We could refactor this later :) Small PR is always better.